### PR TITLE
Enable firestore

### LINKED ISSUE #39: Enable Firestore
update .gitgnore 
App-level build.gradle 
Project-level build.gradle
create Podfile
create Test screen for hook up
Change minSdk for Android to 21
Update yaml file

Add cloud firestore 
Add storage 
Add crashlytics 
Add authentication 

Test build function on iOS and Android

### PR DESCRIPTION
**Pull request by taracell**
Fully developed and tested Crashylitics and Firebase connection.  All associated configuration changes to both Android and iOS are included in this pull request.  Podfile was created.  File test_screen.dart built to crash on launch to test crash, if you want to test the build files remove line 12 from test_screen.dart.  Make sure you put your FireStore files in the right places or the app will not build.

### CODE REVIEW

**Review by DanielQuick — CHANGES_REQUESTED** [review]
I haven't done a thorough review, but these are the issues I found at a glance.

main.dart should not be updated, any async initialization should take place in a splash screen. We don't have one yet, I'll toss one in there shortly after writing this review. You can then pull the changes from the develop branch into this branch and make use of it for Firebase initialization.

Along the same vein, this branch does not require any theming to be added, so that should be removed as well in main.dart

Any test files should be placed in a directory and have a clear filename. I suggest lib/test/ for the directory, and your test_screen.dart could be test_firebase_authentication_page.dart (and also change LoginScreen class to TestFirebaseAuthenticationPage)

The current lib/test_screen.dart also has no functionality in it, it only shows a nasa image and a couple of text widgets. Not too sure what the purpose is.

After you get those changes in I'll do a proper review & build to devices to make sure it all works right.

**Comment by DanielQuick** [comment]
The splash screen is now available on the develop branch.

File: lib/view/page/splash_page.dart [hunk 1]
```diff
@@ -23,6 +25,10 @@ class _SplashPageState extends State<SplashPage> {
       // ensure the onComplete callback cannot be called in the same frame
       widget.onComplete();
     });
+    await Firebase.initializeApp();
+    FirebaseCrashlytics.instance.setCrashlyticsCollectionEnabled(true);
+    /// Pass all uncaught errors from the framework to Crashlytics.
+    FlutterError.onError = FirebaseCrashlytics.instance.recordFlutterError;
```
Comment (DanielQuick) [inline lib/view/page/splash_page.dart]: Firebase should be initialized before the onComplete callback. Should be placed under line 23.

File: pubspec.yaml [hunk 2]
```diff
@@ -12,20 +12,21 @@ dependencies:
   flutter:
     sdk: flutter
 
+  # do not change to cupertino_icons to ^1.0.2 until dart enforces null safety.  This should be next flutter/dart major udpate
   cupertino_icons: ^1.0.0
   provider: ^4.3.2+3
-  get_it: ^5.0.3
+  get_it: ^5.0.4
```
Comment (DanielQuick) [inline pubspec.yaml]: Updating existing dependencies is not in the scope of this branch, unless you absolutely require it.

**Review by DanielQuick — CHANGES_REQUESTED** [review]
Just a couple small changes! After they are made I'll try it out in a build, then we are ready to approve it!

File: pubspec.yaml (hunk 2 above)
Comment (taracell) [reply pubspec.yaml]: Sorry, my add on updated everything...I'll make sure to double check to only update relevant dependencies from now on.

**Review by DanielQuick — APPROVED** [review]

**Pull request merged**